### PR TITLE
Normalize AI-generated commit titles by stripping conventional prefixes and update prompts

### LINKED ISSUE #69: [Commit Message Generation] Avoid duplicate commit types when using z.ai models with simple/advanced prompts
## Summary

`git-acp` can generate commit messages with duplicate commit types when using z.ai models, especially with `--prompt-type advanced` and some `--prompt-type simple` outputs. The AI-generated message already contains a conventional commit prefix, and the later commit classification step prepends another type, producing messages such as `test 🧪: fix ...` or `refactor ♻️: refactor(...) ...`.

## Steps to reproduce

1. Use repo `beecave-homelab/git-acp`.
2. Configure z.ai as the AI provider/model:
   - Base URL: `https://api.z.ai/api/coding/paas/v4`
   - Model: `GLM-4.7-flash`
   - Temperature: `0.3`
   - Context window: `32768`
3. Run auto-group dry-run with advanced prompt type:
   ```bash
   git-acp -o -ag --dry-run -nc -v -ct 32768 -pt advanced
   ```
4. Run auto-group dry-run with simple prompt type:
   ```bash
   git-acp -o -ag --dry-run -nc -v -ct 32768 -pt simple
   ```

## Expected result

Generated commit messages should follow one consistent output format with only one commit type prefix, for example:

```
fix 🐛: Add ffmpeg fallback for audio preloading in diarization
```

or:

```
refactor ♻️: use ffmpeg fallback for audio preloading
```

## Actual result

The final dry-run commit message sometimes contains two commit types/prefixes.

Examples from `--prompt-type advanced`:

```
Would commit with message:
test 🧪: fix diarization: Add ffmpeg fallback for unsupported audio formats
```

```
Would commit with message:
test 🧪: fix 🐛: Add ffmpeg fallback for audio preloading in diarization
```

Examples from `--prompt-type simple`:

```
Would commit with message:
refactor ♻️: refactor(diary): use ffmpeg fallback for audio preloading
```

The logs show the AI output already included a commit type, after which commit classification prepended another type. In one advanced run, the generated message was `fix diarization: ...`, then classification selected TEST from file paths and produced `test 🧪: fix diarization: ...`. In another advanced run, the generated message was already `fix 🐛: ...`, then classification again prepended `test 🧪:`.

The simple prompt also produced a conventional-commit-style message, `refactor(diary): ...`; classification detected refactor from `message_prefix`, then prepended `refactor ♻️:` again.

## Environment

- Repo: beecave-homelab/git-acp
- Command mode: auto-group dry-run
- AI provider: z.ai
- Model: GLM-4.7-flash
- Prompt types tested: advanced, simple
- Context window: 32768
- Flags used: `-o -ag --dry-run -nc -v -ct 32768`

## Frequency

Often with advanced; reproducible at least once with simple.

## Impact

Generated commit messages require manual cleanup before committing. In non-dry-run usage, this could create malformed commit history with duplicate or confusing commit types.

## Evidence

Relevant final messages from the logs:

```
test 🧪: fix diarization: Add ffmpeg fallback for unsupported audio formats
test 🧪: fix 🐛: Add ffmpeg fallback for audio preloading in diarization
refactor ♻️: refactor(diary): use ffmpeg fallback for audio preloading
```

The affected run also shows `Auto Group Complete` with `Successful groups: 2` and `Failed groups: 0`, so the command succeeds even though the generated output format is wrong.

## Suspected cause

The prompt asks the model to generate a conventional commit message, so z.ai sometimes includes the type itself. The downstream commit classification step then prepends its own selected type without normalizing or stripping an existing type prefix.

This appears to affect both prompt styles, so the simple and advanced prompts may need to be further refined or simplified for z.ai models.

## Suggested fix

* Ensure the AI prompt asks for only the subject text when commit classification will prepend the type.
* Or normalize the AI response before applying classification:
  * Strip existing conventional commit prefixes like `fix:`, `fix(scope):`, `fix 🐛:`, `refactor(...):`.
  * Avoid prepending a second type if one is already present.
  * Optionally validate the final message against the configured output format.
* Add tests covering z.ai-style outputs that already include a commit type.

### PR DESCRIPTION
**Pull request by beecave-homelab**
### Motivation
- Prevent duplicate or conflicting commit type prefixes when AI-generated messages already include a conventional prefix. 
- Ensure AI produces title-first output without a type prefix so the workflow can apply a single canonical `CommitType`. 
- Make commit classification and formatting robust to AI-provided prefixes and emoji annotations.

### Description
- Add `strip_conventional_prefix` helper to `git_acp.git.classification` to remove `type`, `type(scope)`, `type emoji`, and `type(scope) emoji` style prefixes from titles. 
- Export `strip_conventional_prefix` from `git_acp.git.__init__` and use it in `GitWorkflow._format_message` to normalize AI-provided titles before applying the selected `CommitType`. 
- Update AI prompt templates in `git_acp/ai/ai_utils.py` to instruct models not to include a conventional commit type prefix in the title. 
- Add and update unit tests covering prefix stripping and workflow formatting in `tests/git/test_classification.py`, `tests/cli/test_workflow.py`, and `tests/ai/test_ai_utils.py`, and update `project-overview.md` to reflect the behavior change.

### Testing
- Ran unit tests for the modified areas including `tests/git/test_classification.py`, `tests/cli/test_workflow.py`, and `tests/ai/test_ai_utils.py`. All tests passed.
- Existing AI prompt and formatting tests were updated and verified to succeed.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_6a0642fad088832b818d5a70b4fc160b)

## Summary by CodeRabbit

* **Refactor**
  * Commit message suggestions and formatting now normalize subjects by removing conventional commit-type prefixes for cleaner titles.

* **Bug Fixes**
  * Improved change detection when no staged changes: message generation and type classification now consider selected files and untracked new files so AI suggestions are more accurate; falls back to a default type when diffs are unavailable.

* **Tests**
  * Expanded tests covering prefix handling and message normalization.

[Review Change Stack](https://app.coderabbit.ai/change-stack/beecave-homelab/git-acp/pull/70?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by beecave-homelab** [comment]
@coderabbitai please review.

**Comment by beecave-homelab** [comment]
@coderabbitai review

**Comment by beecave-orchestrator** [comment]
**APPROVED**

 is a well-structured fix for the duplicate prefix problem, with two items worth noting.

---

What's good:

• The dual approach (prompt-level prevention + code-level stripping) is the right strategy. AI models won't always obey prompt instructions, so the strip_conventional_prefix safety net is essential.
• get_diff now accepts an optional files parameter, cleanly solving the problem of classification being polluted by unrelated unstaged changes.
• _diff_for_untracked_files is a reasonable workaround for the fact that git diff ignores untracked files.
• The graceful fallback to CHORE when get_diff raises GitError is a meaningful improvement -- previously, classification would crash instead of defaulting.
• Tests are thorough and cover the key edge cases.

---

Verdict: Solid PR. The core logic is correct and well-tested.

File: git_acp/ai/ai_utils.py [hunk 1]
```diff
@@ -309,7 +347,27 @@ def get_commit_context(config: GitConfig) -> dict[str, Any]:
...
-            staged_changes = get_diff("unstaged", config)
+            # Scope the unstaged diff to the user's selected files so the AI
+            # only sees changes relevant to the intended commit (important in
+            # dry-run mode where files are never actually staged).
+            selected_files: list[str] | None = None
+            if config and config.files and config.files != ".":
+                import shlex
+
+                selected_files = shlex.split(config.files)
+            staged_changes = get_diff("unstaged", config, files=selected_files)
```
[the remark this answers is not in the document]
Comment (beecave-orchestrator) [reply git_acp/ai/ai_utils.py:361]: Implemented in commits 45af329..ad9d195 — try/except ValueError wrapping shlex.split with GitError re-raise is now in place.

File: git_acp/git/classification.py [hunk 2]
```diff
@@ -479,6 +479,38 @@ def _parse_message_prefix(commit_title: str, config) -> CommitType | None:
...
+        commit_type.name.lower() for commit_type in CommitType
+    )
+    pattern = re.compile(
+        rf"^\s*(?:{commit_type_pattern})\s*(?:\([^)]+\))?\s*[^A-Za-z0-9_]*:\s*(?P<body>.+)$",
+        flags=re.IGNORECASE,
+    )
+    match = pattern.match(title)
+    if not match:
+        return title
+    return match.group("body").lstrip()
```
[the remark this answers is not in the document]
Comment (beecave-orchestrator) [reply git_acp/git/classification.py]: Intentionally not implemented as suggested. The design decision is to **strip** the breaking-change `!` marker during prefix normalization. When `_format_message` replaces the original commit type with the classified type, the breaking indicator is intentionally discarded — the classified type is authoritative and the original `!` from an AI-generated message should not carry through. The test `test_strip_conventional_prefix__breaking_marker` was added to verify this stripping behavior.
